exchanges/orderbook/depth.go

```diff
@@ -0,0 +1,331 @@
+package orderbook
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/thrasher-corp/gocryptotrader/dispatch"
+	"github.com/thrasher-corp/gocryptotrader/log"
+)
+
+// Depth defines a linked list of orderbook items
+type Depth struct {
+	asks
+	bids
+
+	// unexported stack of nodes
+	stack *stack
+
+	// Change of state to re-check depth list
+	wait    chan struct{}
+	waiting uint32
+	wg      sync.WaitGroup
+	wMtx    sync.Mutex
+	// -----
+
+	mux *dispatch.Mux
+	id  uuid.UUID
+
+	options
+	sync.Mutex
```

Comment:
Segregation mainly so routine(s) waiting for a change does not interfere with the main lock until it gets an alert. As for the `wMtx` that refers to waiting or queue not necessarily a 'write'. I think your suggestion about using a RWMutex might cause overhead and slow execution, because there might be a lot of write locking if there are many routines waiting or about to wait.

Reply:
Thanks for clarifying!